tests/queries/0_stateless/02994_merge_tree_mutations_cleanup.sql.j2

```diff
@@ -0,0 +1,14 @@
+drop table if exists data_rmt;
+drop table if exists data_mt;
+
+create table data_rmt (key Int) engine=ReplicatedMergeTree('/tables/{database}/data', '{table}') order by tuple() settings finished_mutations_to_keep=0, merge_tree_clear_old_parts_interval_seconds=1;
+create table data_mt (key Int) engine=MergeTree() order by tuple() settings finished_mutations_to_keep=0, merge_tree_clear_old_parts_interval_seconds=1;
+
+{% for table in ['data_rmt', 'data_mt'] %}
+alter table {{table}} delete where 1;
```

Comment:
In our private CI this failed:
```
2024-03-01 07:40:58 @@ -1,4 +1,4 @@
2024-03-01 07:40:58 -mutations after ALTER for data_rmt	1
2024-03-01 07:40:58 +mutations after ALTER for data_rmt	0
2024-03-01 07:40:58  mutations after cleanup for data_rmt	1
2024-03-01 07:40:58  mutations after ALTER for data_mt	1
2024-03-01 07:40:58  mutations after cleanup for data_mt	1
```
So I think we should wait for the mutation to be executed:
```suggestion
alter table {{table}} delete where 1 settings mutations_sync = 1;
```